### PR TITLE
v0.54.9 - Fixes and ts-scripts improvements

### LINKED ISSUE #1282: Elasticsearch request error when adding field types to data type
when adding new field types to a datatype I get:
```
Request Error
Elasticsearch Error: [mapper_parsing_exception] failed to parse
```
I was adding 3 new fields, 2 are marked as arrays, though I don't think that has anything to do with the error.  I have saved field types as arrays and not seen the error.  

I have not noticed a pattern to this error, it seems random.  I will add to this ticket if I see it again.

Screen Shot 2019-07-30 at 1 10 42 PM

### PR DESCRIPTION
**Pull request by peterdemartini**
- Moves npm and docker build scripts to `ts-scripts`
- Makes ts-scripts more generic and flexible
- Fixes `ui-data-access` Issues
- Use `utils` error parsing vs the depecrated `error_parser` parsing in teraslice (just one file change)
- Fix error handling issue in `elasticsearch-api->_createIndex`
- Dependency and consistency fixes
- Improved performance metrics using `gc-stats` and `event-loop-stats`

Ref #1299 
Resolves #1282

### CODE REVIEW

**Comment by godber** [comment]
Reviewing

File: packages/teraslice/package.json [hunk 1]
```diff
@@ -74,7 +72,12 @@
     },
     "displayName": "Teraslice",
     "terascope": {
+        "main": true,
         "enableTypedoc": false,
         "testSuite": "elasticsearch"
+    },
+    "optionalDependencies": {
```
Comment (godber) [inline packages/teraslice/package.json]: I'm not sure it makes sense to make these optional dependencies.  Doing so makes this effectively a build time choice, but the feature is triggered with a job option.  That would mean users could try to use it, only to discover that support was not provided at build time.

I might feel differently if these packages are particularly huge or something.  If you leave these as optional dependencies, you might want to change your error message in your catch statements.  `Failure to construct gc-stats` should be more along the lines of `Failure to construct gc-stats, gs-stats module may not be installed`.

File: packages/teraslice/test/workers/helpers/configs.js [hunk 2]
```diff
@@ -72,7 +72,8 @@ const newConfig = (options = {}) => {
         lifecycle,
         max_retries: maxRetries,
         operations,
-        performance_metrics: false,
+        // this will give it a 20% it will be enabled
+        performance_metrics: Math.random() < 0.2,
```
Comment (godber) [inline packages/teraslice/test/workers/helpers/configs.js]: This introduces non-determinism into your tests.  I question if it's worth it.

File: packages/teraslice/package.json (hunk 1 above)
Comment (peterdemartini) [reply packages/teraslice/package.json]: Actually our dockerfile ignores optional dependencies so they should be regular dependencies. Fixing it now

**Pull request merged**